q2_coordinates/qtrees.py

```diff
@@ -0,0 +1,204 @@
+    
+
+    ## if selected method of binning is mock
+    #md_edited['country'] = md_df['country']
+
+    df = metadata[['longitude','latitude']]
+
+    #as global or local in function, define all null or this:
+    df[latitude] = pd.to_numeric(df.latitude, errors='coerce')
+    df[longitude] = pd.to_numeric(df.longitude, errors='coerce')
+    
+    #drop nan values (formerly strings) from dataframe
+    df = df.dropna(subset = [longitude, latitude])
+  
+    if df.empty is True:
+        raise ValueError("Latitude and/or Longitude have no numeric values, please check your data.")
+
+    #check if latitude and longitude is castable as float
+    df[latitude] = df[latitude].astype(float)
+    df[longitude] = df[longitude].astype(float)
+
+    #make (0,0) be the bottom right corner for ease of calculation, probably a better way to do this?
+    df[latitude] = df[latitude] + 90
+    df[longitude] = df[longitude] + 180
+    
+    return df
+
+
+class Point():
+    def __init__(self, x, y, sample_id):
+        self.x = float(x)
+        self.y = float(y)
+        self.sample_id = sample_id        
+    
+class Node():
+    def __init__(self, x0, y0, w, h, points, _id):
+        self.x0 = x0
+        self.y0 = y0
+        self.width = float(w)
+        self.height = float(h)
+        self.points = points
+        self.children = []
+        self.id = _id
+
+    def get_width(self):
+        return self.width
+    
+    def get_height(self):
+        return self.height
+    
+    def get_points(self):
+        return self.points
+    
+    def get_id(self):
+        return self.id
+    
+    def set_id(self, _id):
+        self.id = self.id+_id
+
+    
+class QTree():
+    def __init__(self, k, data):
+        self.threshold = k
+        self.points = [Point(x, y, sample_id) for sample_id, x, y in data]
+        self.root = Node(0, 0, 360, 180, self.points, "0")
+
+    def add_point(x, y, sample_id):
+        self.points.append(Point(x, y, sample_id))
+    
+    def get_points(self):
+        return self.points
+    
+    def subdivide(self, samples, threshold):
+        count = 0
+        node_id = ""
+        samples = samples
+        bins  = []
+        
+        samples, bins = recursive_subdivide(self.root, threshold, count, node_id, samples, bins)
+        return samples, bins
+    
+
+def recursive_subdivide(node, k, count, node_id, samples, bins):
+
+    if len(node.points) < k:
+        return
+    elif len(node.points)/k >= len(node.points):
+        raise ValueError("The threshold for subdivision is less than the amount of points in one place,", 
+                         "please chose a larger threshold for division")
+    else:
+        count += 1
+
+    samples["H" + str(count)] = ""
+    
+    w_ = node.width/2
+    h_ = node.height/2    
+    
+    nodes = []
+    sk_nodes = []
+    for i in range(4):
+        #southwest
+        if(i ==0):
+            quad = "3"
```

Comment:
would it be possible for the quadrants described by the code within the ifs and elifs to correspond to the value of `i`?